specification/security/resource-manager/Microsoft.Security/stable/2019-08-01/devices.json

```diff
@@ -0,0 +1,582 @@
+      "get": {
+        "x-ms-examples": {
+          "Get Device": {
+            "$ref": "./examples/Devices/GetDevice.json"
+          }
+        },
+        "tags": [
+          "Device Inventory"
+        ],
+        "description": "Get device.",
+        "operationId": "Device_Get",
+        "parameters": [
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/ApiVersion"
+          },
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/SubscriptionId"
+          },
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/ResourceGroupName"
+          },
+          {
+            "$ref": "#/parameters/SolutionName"
+          },
+          {
+            "$ref": "#/parameters/DeviceId"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/Device"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../common/v1/types.json#/definitions/CloudError"
+            }
+          }
+        }
+      },
+      "patch": {
+        "x-ms-examples": {
+          "Use this method to update device": {
+            "$ref": "./examples/Devices/UpdateDevice.json"
+          }
+        },
+        "tags": [
+          "Device Inventory"
+        ],
+        "description": "Update existing device name, type, operating system, purdue level, device criticality and notes.",
+        "operationId": "Device_Update",
+        "parameters": [
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/ApiVersion"
+          },
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/SubscriptionId"
+          },
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/ResourceGroupName"
+          },
+          {
+            "$ref": "#/parameters/SolutionName"
+          },
+          {
+            "$ref": "#/parameters/DeviceId"
+          },
+          {
+            "$ref": "#/parameters/DeviceData"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Updated",
+            "schema": {
+              "$ref": "#/definitions/Device"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../common/v1/types.json#/definitions/CloudError"
+            }
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "DeviceList": {
+      "description": "List of Devices",
+      "required": [
+        "value",
+        "totalCount"
+      ],
+      "properties": {
```

Comment:
What is not supported? Can you please explain more?

Reply:
Oops, the comment went to the wrong line. it was meant to be for the totalCount property. Returning a "totalCount" is not standard pattern. It can also lead to incorrect value in cases when ARM filters the items from the value array due to insufficient RBAC permissions (yes ARM does post filtering on the collection response).

Reply:
I believe Gaurav is talking about using https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/openapi-authoring-automated-guidelines.md#r2029-pageableoperation

Reply:
Thanks @ravbhatnagar for explaining yes any such field will also cause problems in indexing services like Azure Resource Graph, this pattern should not be used.

Reply:
@ravbhatnagar @chiragg4u This is a proxy resource, so shouldn't RBAC and resource graph be irrelevant?

Reply:
ARG (Resource Graph) does support proxy resources and ARM also applies RBAC on proxy resources. Infact portal is pushing all blades to move to ARG. 
Why totalCount is important?

Reply:
While ARG could support proxy resources, it doesn't by default, and we have no plans to support it for this resource type.
Regarding RBAC, proxy resources just inherit the permissions from their parent tracked resource, right? (That is, you can't set specific roles on a proxy resource).
This field is required so the customer will be able to understand the size of his "device inventory" (potentially could be tens of thousands of devices). These devices are automatically discovered by the service. The customer doesn't create them on it's own.
We happen to have a couple of other proxy resources, similar in their nature to this one, which also use the same 'totalCount' field, and we haven't received similar feedback during their PRs (two separate PRs).

Reply:
@liranc argument this was done in other places is not correct. 
As @ravbhatnagar commented this can cause issues in the user experiences and ARM API reviewers are looking for such scenarios. Ultimately RP's(you) own the experience quality and we would let you take a call on this. Please confirm.

Reply:
I understand and appreciate your feedback 👍
I think at this point we prefer to keep an aligned experience across our APIs. In case it we'll need to make an adjustment with the `totalCount` property, we'll need to make it in other resource types in any case.
Confirmed 🙂